packages/core/src/notifications/controller.ts

```diff
@@ -98,6 +98,16 @@ export class NotificationsController {
             await this.writeState()
             void this.notificationsNode.focusPanel()
         }
+
+        // Process on-receive behavior for newly received notifications that passes rule engine
+        const newlyReceivedToDisplay = [...startUp, ...emergency].filter((n) => this.state.newlyReceived.includes(n.id))
+        if (newlyReceivedToDisplay.length > 0) {
+            await this.notificationsNode.onReceiveNotifications(newlyReceivedToDisplay)
+            // remove displayed notifications from newlyReceived
+            this.state.newlyReceived = this.state.newlyReceived.filter(
+                (id) => !newlyReceivedToDisplay.some((n) => n.id === id)
+            )
+        }
```

Comment:
call `await this.writeState()` here since we are modifying the state.

Reply:
is that really needed? `writeState` doesn't handle the newlyReceived states, and that's the only thing we are modifying here

Reply:
Yes it does, it writes the whole object to global state: `await globals.globalState.update(this.storageKey, this.state)`

You are updating newlyReceived in `fetchNotifications` and then writing the state. But once you clear it here you are never writing the cleared state back. So the newlyReceived state will never be cleared out and if the IDE is restarted it seems like this will re-run for notifications we already received.